src/Client.php

```diff
@@ -202,14 +202,21 @@ private function receive(): stdClass|array|int
         $result = BSON\toPHP(substr($res, 21, $responseLength - 21));
 
         if (property_exists($result, "writeErrors")) {
-            // Throws a Utopia\Mongo\Exception with Code Error
-
+            // Throws Utopia\Mongo\Exception
             throw new Exception(
                 $result->writeErrors[0]->errmsg,
                 $result->writeErrors[0]->code
             );
         }
 
+        if (property_exists($result, 'errmsg')) {
+            // Throws Utopia\Mongo\Exception
+            throw new Exception(
+                'E'.$result->code.' '.$result->codeName.':'.$result->errmsg,
```

Comment:
What is the `codeName`? Do we need to include the code and codeName in the message as we already pass the code as the second parameter?

Reply:
for example:
code = 50 
codeName  = MaxTimeMSExpired
errmsg = operation exceeded time limit

 MaxTimeMS was the value assigned that created the error.

Reply:
Ah I see, I think the read exception `message` should be `$result->errmsg`, the same as we have for a write exception

Reply:
Yep this data is codeName does not exist in writeErrors structure:

{
  ["n"]=>
  int(0)
  ["writeErrors"]=>
  array(1) {
    [0]=>
    object(stdClass)#364 (5) {
      ["index"]=>
      int(0)
      ["code"]=>
      int(11000)
      ["keyPattern"]=>
      object(stdClass)#370 (1) {
        ["_id"]=>
        int(1)
      }
      ["keyValue"]=>
      object(stdClass)#367 (1) {
        ["_id"]=>
        int(999)
      }
      ["errmsg"]=>
      string(87) "E11000 duplicate key error collection: testing.movies index: _id_ dup key: { _id: 999 }"
    }
  }
  ["ok"]=>
  float(1)
}

Reply:
Ah okay, so using `'E'.$result->code.' '.$result->codeName.':'.$result->errmsg,` here is  consistent with `$result->writeErrors[0]->errmsg,` we use for the write exception message? That works for me :+1:

Reply:
As much as possible I am missing a space after ":" I see now :)